snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule_mrcm_constraint_type.groovy

```diff
@@ -240,14 +254,90 @@ def searchRelationshipsWithUnregulatedTypeIds =  {
 		owlMemberExpressionBuilder.filter(SnomedRefSetMemberIndexEntry.Expressions.effectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME))
 	}
 	
-	final Query<String> owlMemberQuery = Query.select(String.class)
+	final Query<String[]> owlMemberQuery = Query.select(String[].class)
 		.from(SnomedRefSetMemberIndexEntry.class)
-		.fields(SnomedRefSetMemberIndexEntry.Fields.ID)
+		.fields(SnomedRefSetMemberIndexEntry.Fields.ID, SnomedRefSetMemberIndexEntry.Fields.OWL_EXPRESSION)
 		.where(owlMemberExpressionBuilder.build())
 		.limit(50_000)
 		.build();
 	
-	searcher.stream(owlMemberQuery).forEach({ hits -> });
+	searcher.stream(owlMemberQuery).forEach({ hits ->
+		hits.each { hit ->
+			def id = hit[0];
+			def owlExpression = hit[1];
+			if (!owlExpression.contains("TransitiveObjectProperty") &&
+				!owlExpression.contains("ReflexiveObjectProperty") &&
+				!owlExpression.contains("SubDataPropertyOf") &&
+				!owlExpression.contains("SubObjectPropertyOf") &&
+				!owlExpression.contains("ObjectPropertyChain")) {
+				//Skip axiom member with no generated axiom relationships
+				issues.add(ComponentIdentifier.of(SnomedReferenceSetMember.TYPE, id))
+			}
+		}
+	});
+}
+
+def searchRelationshipsInUnregulatedDomains =  {
+	String regulatedDomainSpace = Joiner.on(" OR ")
+		.join(FluentIterable.from(domainMembers.values())
+			.transform({ m -> m.getProperties().get(SnomedRf2Headers.FIELD_MRCM_DOMAIN_CONSTRAINT)})
+			.transform({ c -> "(${c})"}));
+				
+	Set<String> unregulatedDomainSpace = getApplicableConcepts("* MINUS (${regulatedDomainSpace})");
+	
+	//Find non-IsA relationships in domains where no MRCM rule is defined
+	ExpressionBuilder relationshipQueryBuilder = Expressions.builder()
+			.filter(SnomedRelationshipIndexEntry.Expressions.active())
+			.filter(SnomedRelationshipIndexEntry.Expressions.sourceIds(unregulatedDomainSpace))
```

Comment:
Is regulated/unregulated the same as sanctioned/unsanctioned? If yes, please rename these variables to the latter.

Reply:
They might, I was trying to make a distinction between cases that break an explicit rule, vs cases that aren't covered by any rules. To me, an attribute that breaks an explicit rule is an unsanctioned attribute, attributes in domains that have no rules defined for them are unregulated.

Reply:
We didn't make that distinction earlier, but I'm not against it if it describes things better. 

Previously any type not covered by an MRCM rule was considered unsanctioned, regardless of whether it was added to a concept living in a regulated or an unregulated region, in exchange for a warning. Breaking the rules is indeed an error (or "sacrilege" if we stick with the terminology adopted for this subject). I don't know if we even offer "clearly illegal" attributes for selection on the UI, but they can be part of an import.

Reply:
We do offer both sanctioned and unsanctioned properties for selection.